tests/41end-to-end-keys/01-upload-key.pl

```diff
@@ -53,14 +53,29 @@
       )->then( sub {
          my ( $content ) = @_;
 
+         log_if_fail( "/query response", $content );
+
          assert_json_keys( $content, "device_keys" );
 
          my $device_keys = $content->{device_keys};
          assert_json_keys( $device_keys, $user->user_id );
 
          my $alice_keys = $device_keys->{ $user->user_id };
          assert_json_keys( $alice_keys, $user->device_id );
+
+         my $alice_device_keys = $alice_keys->{ $user->device_id };
+         assert_json_keys( $alice_device_keys, "unsigned" );
+
+         my $unsigned = $alice_device_keys->{unsigned};
+
+         # display_name should be null by default
+         exists $unsigned->{device_display_name} or
+           die "Expected to get a (null) device_display_name";
+         defined $unsigned->{device_display_name} and
```

Comment:
I dunno. I'd guess that we should handle this in the same way that user displaynames are handled.